mutiny-core/src/hermes.rs

```diff
@@ -180,6 +180,11 @@ impl<S: MutinyStorage> HermesClient<S> {
                                 o.name
                             );
                             *c = (o.name.clone(), true);
+
+                            // if we don't have a lightning address, no need to continue
+                            if o.name.is_none() {
+                                break;
+                            }
```

Comment:
Something seems strange to me about this code. I think this works, but would it make more sense to have `Ok(Some(u))` and `Ok(None)`? It would probably need to do the same `current_address_check_clone` stuff but make be more logical to seperate when the user has a name registered and not.